components/bio-formats-tools/src/loci/formats/tools/ImageConverter.java

```diff
@@ -591,11 +593,22 @@ public boolean testConvert(IFormatWriter writer, String[] args)
         String xml = service.getOMEXML(service.asRetrieve(store));
         OMEXMLMetadataRoot root = (OMEXMLMetadataRoot) store.getRoot();
         IMetadata meta = service.createOMEXMLMetadata(xml);
+        OMEXMLMetadataRoot newRoot = (OMEXMLMetadataRoot) meta.getRoot();
         if (series >= 0) {
-          Image exportImage = new Image(root.getImage(series));
-          Pixels exportPixels = new Pixels(root.getImage(series).getPixels());
+          Image exportImage = newRoot.getImage(series);
+          Pixels exportPixels = newRoot.getImage(series).getPixels();
+
+          if (channel >= 0) {
+            List<Channel> channels = exportPixels.copyChannelList();
+
+            for (int c=0; c<channels.size(); c++) {
+              if (c != channel) {
+                exportPixels.removeChannel(channels.get(c));
+              }
+            }
+          }
+
           exportImage.setPixels(exportPixels);
-          OMEXMLMetadataRoot newRoot = (OMEXMLMetadataRoot) meta.getRoot();
           while (newRoot.sizeOfImageList() > 0) {
```

Comment:
Outside to the scope of this PR, this block could benefit from the new `clearImageList` API introduced in https://github.com/ome/ome-model/pull/185 and which should be available in the next minor release of `ome-xml`.